### PR TITLE
Template updates to reduce sync updates.

### PR DESCRIPTION
**Pull request by kwoodson**
```release-note
NONE
```

Few updates to fix sync updates.

### CODE REVIEW

File: pkg/sync/v5/data/ImageStream.image.openshift.io/openshift/jboss-amq-62.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ spec:
     importPolicy: {}
     name: "1.1"
     referencePolicy:
-      type: ""
+      type: "Source"
```
Comment (charlesakalugwu) [inline pkg/sync/v5/data/ImageStream.image.openshift.io/openshift/jboss-amq-62.yaml]: everywhere 
```suggestion
      type: Source
```

File: pkg/sync/v5/data/StatefulSet.apps/openshift-azure-monitoring/canary.yaml [hunk 2]
```diff
@@ -24,6 +24,7 @@ spec:
           valueFrom:
             fieldRef:
```
Comment (charlesakalugwu) [inline pkg/sync/v5/data/StatefulSet.apps/openshift-azure-monitoring/canary.yaml]:
```
fieldRef:
    apiVersion: v1
    fieldPath: spec.nodeName
```

**Comment by mjudeikis** [comment]
Might be my paranoid eyes, but was under impression that after this got merged:
https://github.com/openshift/openshift-azure/pull/1433

Change you are doing should not be allowed by the content test. I suspect this test is broken. Looking into it,  but for now
/hold
while Im checking

File: pkg/sync/v5/data/StatefulSet.apps/openshift-azure-monitoring/canary.yaml (hunk 2 above)
Comment (kwoodson) [reply pkg/sync/v5/data/StatefulSet.apps/openshift-azure-monitoring/canary.yaml]: Updated.

File: pkg/sync/v5/data/ImageStream.image.openshift.io/openshift/jboss-amq-62.yaml (hunk 1 above)
Comment (kwoodson) [reply pkg/sync/v5/data/ImageStream.image.openshift.io/openshift/jboss-amq-62.yaml]: Updated.

**Comment by kwoodson** [comment]
@charlesakalugwu @Makdaam @mjudeikis  
Please look again.  We need this to go through for 2-3 more consecutive PRs.

**Comment by kwoodson** [comment]
fixes: #1543

**Comment by mjudeikis** [comment]
@kwoodson run this on your PR:
```
go test -v -tags=content ./...
```

I think the solution should be in https://github.com/openshift/openshift-azure/blob/master/pkg/sync/v5/default.go#L42 function, and yaml file directly.

**Comment by y-cote** [comment]
> @kwoodson run this on your PR:
> 
> ```
> go test -v -tags=content ./...
> ```
> I think the solution should be in https://github.com/openshift/openshift-azure/blob/master/pkg/sync/v5/default.go#L42 function, and yaml file directly.

Strange, I ran it on the PR and I don't get failures. Unless I misinterpreted your comment @mjudeikis

**Comment by mjudeikis** [comment]
okey,  looks like upstream is frozen. I think we need to make this "change" inside contentTest, raise an issue in upstream and follow-up when it is solved.

**Comment by Makdaam** [comment]
Fixed default.go and clean.go in #1575 so when you get templates with `go test -v -tags=content ./...` it will remove the empty string value from the yaml file + will remove "Source" from the existing config before comparison.

**Comment by jim-minter** [comment]
closing for #1575

**Pull request closed without merging**